engine/common/requester/engine.go

```diff
@@ -99,18 +101,19 @@ func New(log zerolog.Logger, metrics module.EngineMetrics, net network.Network,
 
 	// initialize the propagation engine with its dependencies
 	e := &Engine{
-		unit:     engine.NewUnit(),
-		log:      log.With().Str("engine", "requester").Logger(),
-		cfg:      cfg,
-		metrics:  metrics,
-		me:       me,
-		state:    state,
-		channel:  channel,
-		selector: selector,
-		create:   create,
-		handle:   nil,
-		items:    make(map[flow.Identifier]*Item),          // holds all pending items
-		requests: make(map[uint64]*messages.EntityRequest), // holds all sent requests
+		unit:          engine.NewUnit(),
+		log:           log.With().Str("engine", "requester").Logger(),
+		cfg:           cfg,
+		metrics:       metrics,
+		me:            me,
+		state:         state,
+		channel:       channel,
+		selector:      selector,
+		create:        create,
+		handle:        nil,
+		items:         make(map[flow.Identifier]*Item),          // holds all pending items
+		requests:      make(map[uint64]*messages.EntityRequest), // holds all sent requests
+		isDispatching: atomic.NewBool(false),
```

Comment:
I find the naming a bit counterintuitive, because it seems to indicate that `isDispatching` is true whenever _any_ dispatch is ongoing. But that is not the case, it is only true when a _forced_ dispatch is running. Would suggest to reflect it in the variable name. 
```suggestion
		forcedDispatchOngoing: atomic.NewBool(false),
```

Reply:
Another thought:
I don't think there is much use running a simple dispatch, if a forced dispatch is already ongoing. We modify the logic for regularly-timed requests here
https://github.com/onflow/flow-go/blob/1f231ce7e3361320018aa65b5a319ef78b075ebe/engine/common/requester/engine.go#L271-L272 
and exit early 
```
case <-ticker.C:
  if e.isDispatching.Load() { 
    return
  }
  dispatched, err := e.dispatchRequest()
```

With this modification, a small time window remains, where a regular dispatch runs into `dispatchRequest()` and then the `unit.Lock`  is already taken by a forced dispatch. But assuming that Forced dispatches potentially take some time to run (which seems to be the case), this extra check might reduce unnecessary dispatches.

Reply:
Good point 👍